statuspage/src/main/java/com/jagrosh/jdautilities/statuspage/Statuspage.java

```diff
@@ -0,0 +1,447 @@
+/*
+ * Copyright 2016-2018 John Grosh (jagrosh) & Kaidan Gustave (TheMonitorLizard)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.jagrosh.jdautilities.statuspage;
+
+import com.jagrosh.jdautilities.commons.async.AsyncFuture;
+import com.jagrosh.jdautilities.commons.async.AsyncTask;
+import com.jagrosh.jdautilities.statuspage.data.*;
+import com.jagrosh.jdautilities.statuspage.endpoints.*;
+import okhttp3.*;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.json.JSONTokener;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+import java.io.IOException;
+import java.io.Reader;
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+
+@Immutable
+public class Statuspage
+{
+    @Nonnull
+    protected static final String URL_API_BASE = "https://status.discordapp.com/api/v2/";
+    @Nonnull
+    protected static final String URL_SUMMARY = URL_API_BASE + "summary.json";
+    @Nonnull
+    protected static final String URL_SERVICE_STATUS = URL_API_BASE + "status.json";
+    @Nonnull
+    protected static final String URL_COMPONENTS = URL_API_BASE + "components.json";
+    @Nonnull
+    protected static final String URL_INCIDENTS_ALL = URL_API_BASE + "incidents.json";
+    @Nonnull
+    protected static final String URL_INCIDENTS_UNRESOLVED = URL_API_BASE + "incidents/unresolved.json";
+    @Nonnull
+    protected static final String URL_SCHEDULED_MAINTENANCES_ALL = URL_API_BASE + "scheduled-maintenances.json";
+    @Nonnull
+    protected static final String URL_SCHEDULED_MAINTENANCES_ACTIVE = URL_API_BASE + "scheduled-maintenances/active.json";
+    @Nonnull
+    protected static final String URL_SCHEDULED_MAINTENANCES_UPCOMING = URL_API_BASE + "scheduled-maintenances/upcoming.json";
```

Comment:
For the `scheduled-maintenances` endpoints and organization's sake, we should have a base url 
```java
    @Nonnull
    protected static final String URL_SCHEDULED_MAINTENANCES_BASE = URL_API_BASE + "scheduled-maintenances";
    @Nonnull
    protected static final String URL_SCHEDULED_MAINTENANCES_ALL = URL_SCHEDULED_MAINTENANCES_BASE + ".json";
    @Nonnull
    protected static final String URL_SCHEDULED_MAINTENANCES_ACTIVE = URL_SCHEDULED_MAINTENANCES_BASE + "/active.json";
    @Nonnull
    protected static final String URL_SCHEDULED_MAINTENANCES_UPCOMING = URL_SCHEDULED_MAINTENANCES_BASE + "/upcoming.json";
```
The same might be worth doing for `incidents` although I'll leave that up to you.